backstop_data/engine_scripts/puppet/price-set/add-price-field-modal.js

```diff
@@ -9,7 +9,6 @@ module.exports = async (engine, scenario, viewport) => {
     await require('./view-and-edit-price-fields')(engine, scenario, viewport);
     await engine.click('#newPriceField');
     await engine.waitFor('.CRM_Price_Form_Field');
-  }, async () => {
-    await page.clickAndWaitForNavigation('a[href="/civicrm/admin/price?action=add&reset=1"]');
   })
+  .catch(async () => {});
```

Comment:
I think you forgot 
```js
.catch(async () => {
  await page.clickAndWaitForNavigation('a[href="/civicrm/admin/price?action=add&reset=1"]');
});
```
here

Reply:
Also, if we now use catch, I think for better readability it should looks like this:

```js
'use strict';
 
const Page = require('../page-objects/crm-page.js');
 
module.exports = async (engine, scenario, viewport) => {
  const page = new Page(engine, scenario, viewport);
   
  await page.checkIfPriceFieldsAreEmpty('Fields')
    .then(async () => {
      await require('./view-and-edit-price-fields')(engine, scenario, viewport);
      await engine.click('#newPriceField');
      await engine.waitFor('.CRM_Price_Form_Field');
    })
    .catch(async () => {
      await page.clickAndWaitForNavigation('a[href="/civicrm/admin/price?action=add&reset=1"]');
    });
```